pkg/pub_worker/lib/src/bin/pana_wrapper.dart

```diff
@@ -210,26 +210,42 @@ Future<({String configKind, String? dartSdkPath, String? flutterSdkPath})>
   bool matchesSdks({
     required Version? dart,
     required Version? flutter,
+    required Version? flutterDartSdk,
     bool allowsMissingVersion = false,
   }) {
     if (!allowsMissingVersion && (dart == null || flutter == null)) {
       return false;
     }
 
-    return sdkMatchesConstraint(
-          sdkVersion: dart,
-          constraint: pubspec.dartSdkConstraint,
-        ) &&
-        sdkMatchesConstraint(
-          sdkVersion: flutter,
-          constraint: pubspec.flutterSdkConstraint,
-        );
+    // Dart SDK
```

Comment:
Should we not split between dart and flutter packages first?

Why does a flutter package need to have its constraint validated against the `sdkVersion: dart`? 

Something like this pseudocode:
```
if (isFlutterPackage) {
  return matches(pubspec.dartSdkConstraint, sdkVersion: flutterDartSdk) &&
    matches(pubspec.flutterSdkConstraint, sdkVersion: flutter)
} else {
  return matches(pubspec.dartSdkConstraint, sdkVersion: dart)
}
```

Reply:
A very lame excuse: we don't have the `isFlutterPackage` logic in `pkg/pub_worker`. But that would be easy to add.

However, because `pana` internally uses both top-level SDKs (e.g. running the top-level Dart SDK for format check), and because this check is cheap, I'd rather keep the current logic that checks Dart SDK constraint against both the top-level and against the Flutter-embedded Dart SDKs. I will usually affect only a small subset of packages, and the difference of sparing one check should not matter much.

Reply:
It is not so much the cost of the extra check. More the correctness I worry about.

Could we not have cases where the flutterDart version matches, but the dart version doesn't match, and this method should really return true, but it doesn't...?

Reply:
> Could we not have cases where the flutterDart version matches, but the dart version doesn't match, and this method should really return true, but it doesn't...?

I would treat this as erring on the safer side.

If there is such a case, we will select the beta or the master channel SDKs, while there is a chance that we would have been fine with the pre-built, the latest stable or the beta (instead of the selected one). We have an extra cost of downloading the SDKs dynamically, otherwise we should be fine with the newer SDK bundle which is for sure is larger than the SDK constraint's minimal SDK version.